stonesoup/initiator/simple.py

```diff
@@ -294,3 +299,214 @@ def initiate(self, detections, timestamp, **kwargs):
+        """Initiates tracks given unassociated measurements
+
+        Parameters
+        ----------
+        detections : set of :class:`~.Detection`
+            A list of unassociated detections
+        timestamp: datetime.datetime
+            Current timestamp
+
+        Returns
+        -------
+        : set of :class:`~.Track`
+            A list of new tracks with an initial :class:`~.GaussianMixture`
+        """
+        tracks = self.initiator.initiate(detections, timestamp, **kwargs)
+
+        for track in tracks:
+            mixture = [
+                TaggedWeightedGaussianState(
+                    state_vector=track.state_vector,
+                    covar=track.covar,
+                    weight=Probability(1),
+                    timestamp=track.timestamp,
+                    tag=[])]
+            track[-1] = GaussianMixtureUpdate(
+                hypothesis=track.hypothesis,
+                components=mixture)
+
+        return tracks
+
+
+class ASDGaussianInitiator(GaussianInitiator):
+    """ASD Gaussian State Initiator class
+
+    Utilising Gaussian Initiator, sample from the resultant track's state
+    to generate an ASD Gaussian State, overwriting with a
+    :class:`~.ASDGaussianState`.
+    """
+
+    initiator: GaussianInitiator = Property(
+        doc="Gaussian Initiator which will be used to generate tracks.")
+    timestamp: datetime.datetime = Property(
+        doc="Starting timestep attributed to the prior state.")
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        # Create prior particle state
+        try:
+            state = self.initiator.prior_state.state_vector
+            covar = self.initiator.prior_state.covar
+
+        except AttributeError:
+            raise AttributeError("No prior state")
+
+        self.prior_state = ASDGaussianState(multi_state_vector=state,
+                                            timestamps=self.timestamp,
+                                            max_nstep=0,
+                                            multi_covar=covar)
+
+    def initiate(self, detections, timestamp, **kwargs):
+        """Initiates tracks given unassociated measurements
+
+        Parameters
+        ----------
+        detections : set of :class:`~.Detection`
+            A list of unassociated detections
+        timestamp: datetime.datetime
+            Current timestamp
+
+        Returns
+        -------
+        : set of :class:`~.Track`
+            A list of new tracks with an initial :class:`~.ASDGaussianState`
+        """
+        tracks = self.initiator.initiate(detections, timestamp, **kwargs)
+
+        for track in tracks:
+            state = track.state_vector
+            covar = track.covar
+            timestamp = track.timestamp
+
+            track[-1] = ASDGaussianStateUpdate(
+                multi_state_vector=state,
+                timestamps=timestamp,
+                max_nstep=0,
```

Comment:
I think this should be a parameter defined on the initiator (like `ensemble_size` below).